presto-orc/src/main/java/com/facebook/presto/orc/reader/RepeatedColumnReader.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.reader;
+
+import com.facebook.presto.orc.QualifyingSet;
+// import com.facebook.presto.orc.stream.BooleanInputStream;
+// import com.facebook.presto.orc.stream.LongInputStream;
+// import com.facebook.presto.spi.block.Block;
+// import com.facebook.presto.spi.type.Type;
+
+// import java.io.IOException;
+import java.util.Arrays;
+import java.util.OptionalInt;
+
+import static com.google.common.base.Verify.verify;
+
+abstract class RepeatedColumnReader
+        extends NullWrappingColumnReader
+{
+    // Starting offset of each result in the element reader's Block.
+    protected int[] elementOffset;
+    // Length of each row in the input QualifyingSet.
+    protected int[] elementLength;
+    // Start of each row of inputQualifyingSet in the inner  data.
+    protected int[] elementStart;
+
+    // Used for compactValues of repeated content.
+    protected int[] innerSurviving;
+    protected int numInnerSurviving;
+    protected int innerSurvivingBase;
+
+    // Number of rows of nested content read. This is after applying any pushed down filters.
+    protected long numNestedRowsRead;
+    // Number of arrays/maps read after applying pushed down filters.
+    protected long numContainerRowsRead;
+
+    RepeatedColumnReader()
+    {
+        super(OptionalInt.empty());
+    }
+
+    protected int getInnerPosition(int position)
+    {
+        return elementOffset[position];
+    }
+
+    protected void computeInnerSurviving(int[]surviving, int base, int numSurviving)
```

Comment:
Prior comment:

---
This is method is not used outside of this class. Make it private.